table.cc

```diff
@@ -2344,3 +2346,39 @@ table::as_mutation_source_excluding(std::vector<sstables::shared_sstable>& ssts)
         return this->make_reader_excluding_sstables(std::move(s), std::move(permit), ssts, range, slice, pc, std::move(trace_state), fwd, fwd_mr);
     });
 }
+
+future<> table::update_repair_time(dht::token_range range, gc_clock::time_point repair_time) {
+    _repair_time_map += std::make_pair(locator::token_metadata::range_to_interval(range), repair_time);
+    return make_ready_future<>();
+}
+
+gc_clock::time_point
+table::get_repair_based_tombstone_gc_before(const dht::decorated_key& dk) const {
+    // FIXME: Make propagation_delay tunable
+    constexpr auto propagation_delay = std::chrono::seconds(3600);
+    if (!use_repair_based_tombstone_gc()) {
+        auto repair_timestamp = gc_clock::now();
+        auto repair_based_tombstone_gc_before = saturating_subtract(repair_timestamp, propagation_delay);
+        tlogger.trace("Get repair based gc time for ks={}, table={}, dk={}, repair_timestamp={}, propagation_delay={}, repair_based_tombstone_gc_before={}, do not use repair_based_tombstone_gc",
+                schema()->ks_name(), schema()->cf_name(), dk, repair_timestamp, propagation_delay.count(), repair_based_tombstone_gc_before);
+        return repair_based_tombstone_gc_before;
```

Comment:
I wonder if this patch will make https://github.com/scylladb/scylla/issues/1745 significantly more likely

Reply:
probably. because now a tombstone can be purged as soon as its associated has been repaired. no longer the default of 10 days. so users out there which repair significantly often but uses the default of 10d are potentially vulnerable to #1745 once this patch is merged. I will work on fixing it soon.

Reply:
Thanks. BTW, In my least version, we still use the old gc_grace_seconds by default. User needs to turn repair based gc explicitly.

Reply:
> In my least version, we still use the old gc_grace_seconds by default.

You mean in the version which is yet to be pushed? Not in the current version of this PR?

Reply:
Yes. See v4 below.